include/abb_librws/rws_common.h

```diff
@@ -79,6 +79,16 @@ struct XMLAttribute
   std::string value;
 };
 
+/**
+ * \brief Returns all children of the given node that have the specifed attribute (name and value).
+ *
+ * \param p_root for the root node to search.
+ * \param attribute specifying the XML attribute (name and value) that the children should have.
+ *
+ * \return std::vector<Poco::XML::Node*> with the children (empty if none are found, or if the root node is nullptr).
+ */
+std::vector<Poco::XML::Node*> xmlFindNodes(Poco::XML::Node* p_root, const XMLAttribute& attribute);
```

Comment:
If this function is not expected to change anything in `p_root`, then perhaps we should make it a const ptr and accept the `Node` also only as `const`?

```suggestion
std::vector<Poco::XML::Node*> xmlFindNodes(Poco::XML::Node const* const p_root, const XMLAttribute& attribute);
```

I don't know whether `Poco::XML::NodeIterator` accepts that though.

Related question: perhaps pass a `const&` instead? I realise you work with pointers elsewhere, but passing `const&` may actually be better.

Reply:
> I don't know whether `Poco::XML::NodeIterator` accepts that though.

I just tested, and it didn't accept it.

> Related question: perhaps pass a `const&` instead? I realise you work with pointers elsewhere, but passing `const&` may actually be better.

I just created https://github.com/ros-industrial/abb_librws/issues/103 to track this.